bus-mapping/src/circuit_input_builder/input_state_ref.rs

```diff
@@ -473,8 +473,41 @@ impl<'a> CircuitInputStateRef<'a> {
         if !found {
             return Err(Error::AccountNotFound(sender));
         }
-        let sender_balance_prev = sender_account.balance;
-        let sender_balance = sender_account.balance - value - fee;
+        let mut sender_balance_prev = sender_account.balance;
+        debug_assert!(
+            sender_account.balance >= value + fee,
+            "invalid amount balance {:?} value {:?} fee {:?}",
+            sender_balance_prev,
+            value,
+            fee
+        );
+        if !fee.is_zero() {
```

Comment:
I think we should have the update even `fee` is zero? Otherwise the `rw_counter` won't match

Reply:
I see. Please let me fix the argument `fee` of `transfer_with_fee` from type `Word` to `Option<Word>`. Since the below `transfer` function calls [this transfer_with_fee](https://github.com/privacy-scaling-explorations/zkevm-circuits/blob/main/bus-mapping/src/circuit_input_builder/input_state_ref.rs#L514).

Reply:
ah i see, using `Option<Word>` should be good.